src/os_crypto/hmac/hmac.c

```diff
@@ -32,8 +32,8 @@ int OS_HMAC_SHA1_Str(const char *key, const char *text, os_sha1 output)
     key_length = strlen(key);
     text_length = strlen(text);
 
+    os_sha1 sha_key;
     if (key_length > HMAC_SHA1_BLOCKSIZE){
```

Comment:
When this condition is true, the parameter "key" returns an invalid pointer that points to a non existing local variable "sha_key". This could be a problem if the calling function tries to use it.